core/services/request/Response.php

```diff
@@ -44,6 +44,7 @@ class Response implements ResponseInterface, ReservedInstanceInterface
     public function __construct()
     {
         $this->terminateRequest(false);
+        $this->registerValidStyles();
```

Comment:
@tn3rb I'm pretty sure that you might know a better place to register this filter. I will change its location after receiving your suggestion.

Reply:
Does this only need to run on admin UI requests ?
Or will it need to run on Frontend UI requests?

i'm saying "UI" requests because this is pertaining to "css styles" which would net be required on a "non-UI" requests such as an AJAX request.

Anyways, it could likely go into one or more of the following depending on where it is needed:

- `EventEspresso\core\domain\entities\routing\handlers\admin\AdminRoute()`
- `EventEspresso\core\domain\entities\routing\handlers\frontend\FrontendRequests::requestHandler()`
- `EventEspresso\core\domain\entities\routing\handlers\frontend\ShortcodeRequests()`
- `EventEspresso\core\domain\entities\routing\handlers\shared\AssetRequests::requestHandler()`
- `EventEspresso\core\domain\entities\routing\handlers\shared\RegularRequests::requestHandler()`

the AssetRequests class might be the best bet out of all of those as it runs on nearly all request types

Reply:
Actually it should be called anywhere we generate HTML content with inline styles.
I assume frontend and backend + AJAX if we generate html content on AJAX calls and returns the output as part of response.

BTW, those classes are not available in `master` and `fix/kses-3` branches.
Please check the current branch and let me know the best place to put it.

Reply:
# 🤦🏻 

sry about that, just throw that filter into `EE_System::initialize()` for now then and I'll deal with it later